ci/do_ci.sh

```diff
@@ -39,39 +39,52 @@ function bazel_with_collection() {
   collect_build_profile $1
 }
 
-function bazel_release_binary_build() {
-  echo "Building..."
-  bazel build ${BAZEL_BUILD_OPTIONS} -c opt //source/exe:envoy-static
-  collect_build_profile release_build
-  # Copy the envoy-static binary somewhere that we can access outside of the
-  # container.
+function cp_binary_for_outside_access() {
+  DELIVERY_LOCATION="$1"
   cp -f \
     "${ENVOY_SRCDIR}"/bazel-bin/source/exe/envoy-static \
-    "${ENVOY_DELIVERY_DIR}"/envoy
+    "${ENVOY_DELIVERY_DIR}"/${DELIVERY_LOCATION}
+}
 
+function cp_binary_for_image_build() {
   # TODO(mattklein123): Replace this with caching and a different job which creates images.
-  echo "Copying release binary for image build..."
-  mkdir -p "${ENVOY_SRCDIR}"/build_release
-  cp -f "${ENVOY_DELIVERY_DIR}"/envoy "${ENVOY_SRCDIR}"/build_release
-  mkdir -p "${ENVOY_SRCDIR}"/build_release_stripped
-  strip "${ENVOY_DELIVERY_DIR}"/envoy -o "${ENVOY_SRCDIR}"/build_release_stripped/envoy
+  echo "Copying binary for image build..."
+  mkdir -p "${ENVOY_SRCDIR}"/build_$1
+  cp -f "${ENVOY_DELIVERY_DIR}"/envoy "${ENVOY_SRCDIR}"/build_$1
+  mkdir -p "${ENVOY_SRCDIR}"/build_$1_stripped
+  strip "${ENVOY_DELIVERY_DIR}"/envoy -o "${ENVOY_SRCDIR}"/build_$1_stripped/envoy
 }
 
-function bazel_debug_binary_build() {
+function bazel_binary_build() {
+  BINARY_TYPE=$1
+  if [[ "${BINARY_TYPE}" == "release" ]]; then
+    COMPILE_TYPE="opt"
+  elif [[ "${BINARY_TYPE}" == "debug" ]]; then
+    COMPILE_TYPE="dbg"
+    BINARY_SUFFIX="-debug"
+  elif [[ "${BINARY_TYPE}" == "sizeopt" ]]; then
+    COMPILE_TYPE="opt"
```

Comment:
Do we still need this? I guess it can't hurt and makes the script simpler but it's a bit redundant given that you added it to .bazelrc

Reply:
It is redundant, but I left it in to keep this function simple.

Reply:
Yup sounds good. Maybe just add a comment to that effect?